### PR TITLE
create a transaction uuid for a sale

### LINKED ISSUE #541: Cannot make a repeat donation (one-time sale)
If you enter a donation twice, the sale API raises an exception about a duplicate `uuid`, which also appears to be blank. This is confusing to me because I thought the `uuid` would be auto-generated. Is this only happening for me?

```
Traceback (most recent call last):
  File "/opt/rh/python27/root/usr/lib64/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 189, in __call__
    response = self.get_response(request)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 218, in get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/rest_framework/views.py", line 399, in dispatch
    response = self.handle_exception(exc)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/rest_framework/views.py", line 396, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/rest_framework/decorators.py", line 49, in handler
    return func(*args, **kwargs)
  File "/srv/solitude/lib/brains/views/sale.py", line 40, in create
    uid_support=result.transaction.id
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/query.py", line 348, in create
    obj.save(force_insert=True, using=self.db)
  File "/srv/solitude/solitude/base.py", line 293, in save
    super(Model, self).save(*args, **kw)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/base.py", line 762, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/base.py", line 846, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/base.py", line 885, in _do_insert
    using=using, raw=raw)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/query.py", line 920, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 974, in execute_sql
    cursor.execute(sql, params)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/utils.py", line 97, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
IntegrityError: (1062, "Duplicate entry '' for key 'uuid'")
````

### PR DESCRIPTION
**Pull request by andymckay**
* fixes #541 
* moves the shorter id creation into the model to make it easier
* adds regression test and a bit of logging just for fun

### CODE REVIEW

File: lib/brains/tests/test_sale.py [hunk 1]
```diff
@@ -95,3 +95,15 @@ def test_pay_method(self):
...
+        self.mocks['sale'].create.return_value = successful_method()
+        seller, seller_product = create_seller()
+
+        data = {'amount': 5, 'nonce': 123, 'product_id': 'brick'}
+        res = self.client.post(self.url, data=data)
+        eq_(res.status_code, 200, res.content)
+
+        self.mocks['sale'].create.return_value = successful_method(id='two-id')
+        res = self.client.post(self.url, data=data)
+        eq_(res.status_code, 200, res.content)
```
Comment (kumar303) [inline lib/brains/tests/test_sale.py]: I don't think this test is really necessary. You could just add an assertion to the creation method that makes sure the `uuid` isn't empty.

File: lib/brains/views/sale.py [hunk 2]
```diff
@@ -39,12 +39,19 @@ def create(request):
         type=constants.TYPE_PAYMENT,
         uid_support=result.transaction.id
     )
+    our_transaction.uuid = our_transaction.create_short_uid()
+    our_transaction.save()
+    log.info('Transaction created: {}'.format(our_transaction.pk))
```
Comment (kumar303) [inline lib/brains/views/sale.py]: :100: yay logs

**Comment by kumar303** [comment]
r+wc

File: lib/brains/tests/test_sale.py (hunk 1 above)
Comment (andymckay) [inline lib/brains/tests/test_sale.py]: I don't think its a huge burden and something else could crop up other than `uuid` to cause it to fail. I've got another test like it, I think I'll keep it for now.

**Pull request merged**